### PR TITLE
[#149] Add spacing screen in demo app

### PR DESCRIPTION
**Pull request by ludovic35**
_Note: Please transform `- [ ]` into `- (NA)` in the description when things are not applicable_

### Types of change

- New feature (non-breaking change which adds functionality)

#### Contribution

- [x] I have read the [contributing guidelines](https://github.com/Orange-OpenSource/ouds-ios/blob/develop/.github/CONTRIBUTING.md)

#### Accessibility

- [x] My change follows accessibility good practices

#### Design

- [x] My change respects the design guidelines of _Orange Unified Design System_

#### Development

- [x] My change follows the [developer guide](https://github.com/Orange-OpenSource/ouds-ios/blob/develop/.github/DEVELOP.md)
- (NA)  I have added unit tests to cover my changes _(optional)_

#### Documentation

- [x] My change introduces changes to the documentation and/or I have updated the documentation accordingly

### Checklist (for Core Team only)

- [x] The evolution have been tested and the project builds for iPhones and iPads
- [x] Code review has been done by reviewers according to [CODEOWNERS file](https://github.com/Orange-OpenSource/ouds-ios/blob/develop/.github/CODEOWNERS)
- (NA) Design review has been done
- (NA) Accessibiltiy review has been done
- (NA) Q/A team has tested the evolution
- [x] Documentation has been updated if relevant
- [x] Internal files have been updated if relevant (like CONTRIBUTING, DEVELOP, THIRD_PARTY, CONTRIBUTORS, NOTICE)
- [x] CHANGELOG has been updated respecting [keep a changelog rules](https://keepachangelog.com/en/1.0.0/) and reference the issues

### CODE REVIEW

File: NOTICE.txt [hunk 1]
```diff
@@ -71,6 +71,7 @@ Any use or displaying shall constitute an infringement under intellectual proper
 ./Showcase/Showcase/Resources/Assets.xcassets/ic_info.imageset/ic_info.svg
 ./Showcase/Showcase/Resources/Assets.xcassets/Illustrations/il_empty_screen.imageset/il_empty_screen.svg
 ./Showcase/Showcase/Resources/Assets.xcassets/Tokens/ic_border.imageset/ic_border.svg
+./Showcase/Showcase/Resources/Assets.xcassets/Tokens/ic_dimension.imageset/ic_dimension.svg
```
Comment (pylapp) [inline NOTICE.txt:74]: Je note que le fichier s'appelle **ic_spacing.svg** , je pense que le fichier NOTICE n'est pas bon ou alors qu'il faut renommer le fichier

File: OUDS/Core/Tokens/SemanticTokens/Sources/Multiples/MultipleSpacingTokens.swift [hunk 2]
```diff
@@ -41,4 +42,11 @@ public final class MultipleSpacingTokens: NSObject {
         guard let other = object as? MultipleSpacingTokens else { return false }
         return self.compact == other.compact && self.regular == other.regular
     }
+
+    /// Returns the right dimension according to the `userInterfaceSizeClass`.
+    /// - Parameter userInterfaceSizeClass: The user interface size class (Could be horizontal or vertical)
```
Comment (pylapp) [inline OUDS/Core/Tokens/SemanticTokens/Sources/Multiples/MultipleSpacingTokens.swift:47]: Cette ligne
```swift
/// - Parameter userInterfaceSizeClass: The user interface size class (Could be horizontal or vertical)
```
n'est pas cohérente. Quand je lis le code, `UserInterfaceSizeClass` ne me semble pas avoir de valeur "horizontal" et "vertical".

File: Showcase/Showcase/Pages/Utils/ShowcaseVariantElement.swift [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+//
+// This software is distributed under the MIT license,
+// the text of which is available at https://opensource.org/license/MIT/
+// or see the "LICENSE" file for more details.
+//
+// Authors: See CONTRIBUTORS.txt
+// Software description: A SwiftUI components library with code examples for Orange Unified Design System
+//
+
+import SwiftUI
```
Comment (pylapp) [inline Showcase/Showcase/Pages/Utils/ShowcaseVariantElement.swift:14]: Les _imports_ ne sont pas ordonnés alphabétiquement

**Review by pylapp — CHANGES_REQUESTED** [review]
Des petites choses à revoir :)

File: NOTICE.txt (hunk 1 above)
Comment (ludovic35) [reply NOTICE.txt:74]: oups

File: Showcase/Showcase/Pages/Utils/ShowcaseVariantElement.swift (hunk 3 above)
Comment (ludovic35) [reply Showcase/Showcase/Pages/Utils/ShowcaseVariantElement.swift]: swift lint ne la pas vu, étrange

File: OUDS/Core/Tokens/SemanticTokens/Sources/Multiples/MultipleSpacingTokens.swift [hunk 4]
```diff
@@ -41,4 +42,11 @@ public final class MultipleSpacingTokens: NSObject {
         guard let other = object as? MultipleSpacingTokens else { return false }
         return self.compact == other.compact && self.regular == other.regular
     }
+
+    /// Returns the right dimension according to the `userInterfaceSizeClass`.
+    /// - Parameter userInterfaceSizeClass: The user interface size class (Could be the horizontal or the vertical size class)
+    /// - Returns: The right dimension raw token
+    public func dimension(for userInterfaceSizeClass: UserInterfaceSizeClass) -> DimensionRawToken {
+        userInterfaceSizeClass == .compact ? compact : regular
```
Comment (pylapp) [inline OUDS/Core/Tokens/SemanticTokens/Sources/Multiples/MultipleSpacingTokens.swift:50]: Comment ça peut compiler ? Je n'ai pas testé mais si j'en crois la doc (https://developer.apple.com/documentation/swiftui/userinterfacesizeclass) `UserInterfaceSizeClass` n'a pas de valeur _compact_

File: Showcase/Showcase/Pages/Utils/ShowcaseVariantElement.swift [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+    @Environment(\.theme) private var theme
+    @Environment(\.colorScheme) private var colorScheme
+
+    // MARK: Stored properties
+
+    let elements: [ShowcaseElement]
+
+    // MARK: Initializer
+
+    init(elements: [ShowcaseElement]) {
```
Comment (pylapp) [inline Showcase/Showcase/Pages/Utils/ShowcaseVariantElement.swift:29]: L'init ne sert à rien ici, il faut l'enlever

**Review by pylapp — CHANGES_REQUESTED** [review]

**Review by pylapp — APPROVED** [review]
LGTM

**Pull request merged**